### PR TITLE
Use service instead of principal in MetricsInterceptor

### PR DESCRIPTION
**Pull request by sfavron**
When recording the caller for http_request_latency_ms, we should only be
recording the calling service name, if it exists, not a username. Since
`caller.get()?.principal` will return service or username, instead use
just `.service`. Having the username tagged on these metrics causes
cardinality explosion.

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt [hunk 1]
```diff
@@ -20,10 +20,10 @@ internal class MetricsInterceptor internal constructor(
     val (elapsedTime, result) = timed { chain.proceed(chain.httpCall) }
 
     val elapsedTimeMillis = elapsedTime.toMillis().toDouble()
-    val callingPrincipal = caller.get()?.principal ?: "unknown"
+    val callingService = caller.get()?.service ?: "unknown"
```
Comment (mightyguava) [inline misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt:23]: we should still tag user as something, like `<user>`

Comment (clarkds14) [inline misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt:23]: add a test that creates a request with `USER_HEADER` set and show that unknown is set in the metric?

**Review by clarkds14 — CHANGES_REQUESTED** [review]

File: misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt [hunk 2]
```diff
@@ -20,10 +20,10 @@ internal class MetricsInterceptor internal constructor(
...
 
     val statusCode = chain.httpCall.statusCode
-    requestDuration.record(elapsedTimeMillis, actionName, callingPrincipal, statusCode.toString())
+    requestDuration.record(elapsedTimeMillis, actionName, callingService, statusCode.toString())
```
Comment (clarkds14) [inline misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt:26]: add a test that creates a request with USER_HEADER set and show that unknown is set in the metric?

File: misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt [hunk 3]
```diff
@@ -20,7 +20,11 @@ internal class MetricsInterceptor internal constructor(
     val (elapsedTime, result) = timed { chain.proceed(chain.httpCall) }
 
     val elapsedTimeMillis = elapsedTime.toMillis().toDouble()
-    val callingPrincipal = caller.get()?.principal ?: "unknown"
+    val callingPrincipal = when {
+      caller.get()?.service != null -> caller.get()?.service!!
+      caller.get()?.user != null -> "user"
```
Comment (mightyguava) [inline misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt:25]: This should have some special characters in there to indicate it's "special" and not conflict with a service potentially named "user", hence why I suggested `<user>`

Comment (sfavron) [reply misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt:25]: 👍🏻 will update

**Review by clarkds14 — APPROVED** [review]

**Review by mightyguava — APPROVED** [review]

**Pull request merged**